singularity-eos/base/fast-math/logs.hpp

```diff
@@ -78,76 +62,166 @@
 namespace singularity {
 namespace FastMath {
 
-// Integer Aliased versions. Likely not as portable
-// TODO(JMM): Add single-precision versions
+template <typename T>
+PORTABLE_FORCEINLINE_FUNCTION auto sgn(const T &x) {
+  return (T(0) < x) - (x < T(0));
+}
+
 PORTABLE_FORCEINLINE_FUNCTION
-auto as_long_int(double f) { return *reinterpret_cast<long long int *>(&f); }
+auto as_int(double f) { return *reinterpret_cast<std::int64_t *>(&f); }
 PORTABLE_FORCEINLINE_FUNCTION
-auto as_double(long long int i) { return *reinterpret_cast<double *>(&i); }
+auto as_double(std::int64_t i) { return *reinterpret_cast<double *>(&i); }
```

Comment:
Would it not be more easier to write

```c++
template<class From, class To, class = std::enable_if<some_type_restrictions<From,To>::value>::type>
To as(From f) { return *reinterpret_cast<To*>(&f); }
```

Reply:
ah, you're thinking to enforce type width. Yes, that's a good idea.

Reply:
gonna leave this for now just for simplicity. But will add a TODO.